astropy/cosmology/_src/flrw/base.py

```diff
@@ -1220,27 +1222,39 @@ def _integral_age(self, z, /):
         """
         return quad(self._lookback_time_integrand_scalar, z, inf)[0]
 
+    # ---------------------------------------------------------------
+    # Comoving distance
+
+    @overload
+    def comoving_distance(self, z: InputT) -> Quantity: ...
+
+    @overload
+    def comoving_distance(self, z: InputT, z2: InputT) -> Quantity: ...
+
     @deprecated_keywords("z", since="7.0")
-    def comoving_distance(self, z):
-        """Comoving line-of-sight distance in Mpc at a given redshift.
+    def comoving_distance(self, z: InputT, z2: InputT | None = None) -> Quantity:
```

Comment:
hum, `z2` should probably be restricted to positional-only from the get go, though while `z` is still accepted as keyword, it seems a bit tricky to implement (but not impossible !). One would need to accept arbitrary `*args`, then add logic to ensure it's exactly 0 on 1-lenght. Do you think it's worth it ?

Reply:
Kind of not 😆. It might be easier to just add a `@deprecated_keywords("z2", since="7.1")` and then sequentially make `z` positional-only  then `z2` positional-only.

Reply:
If `z2` is never *silently* allowed as keyword I don't think we need to change the API more than once; we could end both deprecations at the same point (but it'd need to be astropy 9.0...)

Reply:
> but it'd need to be astropy 9.0...

... so if you're planning on ending the cycle on astropy 8.0 for the first argument, I'm on board with your plan

Reply:
Yes. v8 for `z`, I guess `v9` for z2.